### PR TITLE
Added test for catalogue manager

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: Tests/Unit/Catalogue/CatalogueManagerTest.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        $this->assertCount(2, $messages);
+    }
+
+    public function testFindMessagesNoMetadata()
+    {
+        $manager = new CatalogueManager();
+        $catA = new MessageCatalogue('en', ['messages' => ['a' => 'aTrans', 'b' => 'bTrans']]);
+        $catB = new MessageCatalogue('fr', ['messages' => ['a' => 'aTransFr', 'c' => 'cTransFr']]);
+        $manager->load([$catA, $catB]);
```
Comment (rvanlaak) [inline Tests/Unit/Catalogue/CatalogueManagerTest.php]: What about moving the loading to the `setUp` method, and making the `CatalogueManager` a class field? Seems the initialization is the same in almost every test method.

File: Model/Metadata.php [hunk 2]
```diff
@@ -136,6 +140,9 @@ public function getAllInCategory($category)
     {
         $data = [];
         foreach ($this->notes as &$note) {
```
Comment (Nyholm) [inline Model/Metadata.php]: You are correct. Thank you

File: Tests/Unit/Catalogue/CatalogueManagerTest.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+{
+    public function testGetMessages()
+    {
+        $manager = new CatalogueManager();
+        $catA = new MessageCatalogue('en', ['messages' => ['a' => 'aTrans', 'b' => 'bTrans']]);
+        $catB = new MessageCatalogue('fr', ['messages' => ['a' => 'aTransFr', 'c' => 'cTransFr']]);
+        $manager->load([$catA, $catB]);
+        $messages = $manager->getMessages('en', 'messages');
+
+        $this->assertCount(2, $messages);
```
Comment (Nyholm) [inline Tests/Unit/Catalogue/CatalogueManagerTest.php]: Sure, but I do not understand. 
`assertNotCount(3, ...` of what?

File: Tests/Unit/Catalogue/CatalogueManagerTest.php (hunk 1 above)
Comment (Nyholm) [inline Tests/Unit/Catalogue/CatalogueManagerTest.php]: I know it is a lot of duplication or "almost duplication". I only see it possible to move `$manager = new CatalogueManager();` to setUp. Is it worth it? Or can we move more?

File: Tests/Unit/Catalogue/CatalogueManagerTest.php (hunk 3 above)
Comment (rvanlaak) [inline Tests/Unit/Catalogue/CatalogueManagerTest.php]: By adding one more message to the `fr` catalog, and then asserting `assertNotCount(3, ...` to make sure it does not return the French messages.

File: Tests/Unit/Catalogue/CatalogueManagerTest.php (hunk 1 above)
Comment (rvanlaak) [inline Tests/Unit/Catalogue/CatalogueManagerTest.php]: It seems like all instantiations of `MessageCatalogue` could be moved as well, as `setMetadata` can be called by the test afterwards.

**Pull request merged**